File: android-room-sample-master/app/src/javaCommon/java/com/emreeran/android/roomsample/db/vo/PostWithUserCommentsLikes.java
package com.emreeran.android.roomsample.db.vo;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Relation;

import com.emreeran.android.roomsample.db.entity.Comment;
import com.emreeran.android.roomsample.db.entity.Like;
import com.emreeran.android.roomsample.db.entity.Post;
import com.emreeran.android.roomsample.db.entity.User;

import java.util.List;

/**
 * Created by Emre Eran on 25.04.2018.
 */
public class PostWithUserCommentsLikes {
    @Embedded
    public final Post post;

    @Embedded(prefix = "user_")
    public final User user;

    @Relation(parentColumn = "id", entityColumn = "postId", entity = Comment.class)
    public List<Comment> comments;

    @Relation(parentColumn = "id", entityColumn = "post_id", entity = Like.class)
    public List<Like> likes;

    public PostWithUserCommentsLikes(Post post, User user) {
        this.post = post;
        this.user = user;
    }

    @Override
    public String toString() {
        return "PostWithUserCommentsLikes{" +
                "postWithUser=" + post.toString() +
                ", user=" + user.toString() +
                ", comments=" + comments +
                ", likes=" + likes +
                '}';
    }
}

File: android-room-sample-master/app/src/kotlinCommon/java/com/emreeran/android/roomsample/db/vo/PostWithUserCommentsLikes.kt
package com.emreeran.android.roomsample.db.vo

import android.arch.persistence.room.Embedded
import android.arch.persistence.room.Relation
import com.emreeran.android.roomsample.db.entity.Comment
import com.emreeran.android.roomsample.db.entity.Like
import com.emreeran.android.roomsample.db.entity.Post
import com.emreeran.android.roomsample.db.entity.User

/**
 * Created by Emre Eran on 26.05.2018.
 */
data class PostWithUserCommentsLikes(
        @Embedded val post: Post,
        @Embedded(prefix = "user_") val user: User
) {
    @Relation(parentColumn = "id", entityColumn = "postId", entity = Comment::class)
    var comments: List<Comment>? = null

    @Relation(parentColumn = "id", entityColumn = "postId", entity = Like::class)
    var likes: List<Like>? = null
}